oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -614,9 +640,12 @@ impl Runtime {
         &self,
         node_id: NodeId,
         readers: &[Handle],
-    ) -> Result<Vec<ChannelReadStatus>, OakStatus> {
-        self.validate_handles_access(node_id, readers)?;
-        self.validate_can_read_from_channels(node_id, readers)?;
+    ) -> Result<Vec<ChannelReadStatus>, ErroneousChannelStatus> {
+        self.validate_handles_access(node_id, readers)
+            .map_err(|e| ErroneousChannelStatus::Err(e))?;
+        self.validate_can_read_from_channels(node_id, readers)
+            .map_err(|e| ErroneousChannelStatus::Err(e))?;
+        self.validate_all_channels_are_valid(node_id, readers)?;
```

Comment:
I don't think this logic should be duplicated, ideally the logic in the main loop should take care of this, whether it happens at start or asynchronously?

Reply:
@tiziano88  WDYT?

Reply:
I think I just meant earlier in the loop, but I'm not even sure if that works, or if it needs to happen after adding all the waiters etc.